lib/core/errors.js

```diff
@@ -217,6 +217,15 @@ class SecureProxyConnectionError extends UndiciError {
   }
 }
 
+class AgentMaxOriginsReachedError extends UndiciError {
+  constructor (message) {
+    super(message)
+    this.name = 'AgentMaxOriginsReachedError'
+    this.message = message || 'Agent has reached the maximum allowed origins'
+    this.code = 'UND_ERR_MAX_ORIGINS'
```

Comment:
I guess this shows also that it is maybe a more common error?

Maybe rename it to `MaxOriginsReachedError` and the code 

```suggestion
    this.code = 'UND_ERR_MAX_ORIGINS_REACHED'
```